src/Home/useLocationIsNeededStatus.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from "react"
+import { Platform } from "react-native"
+
+import { doesDeviceSupportLocationlessScanning } from "../gaen/nativeModule"
+
+export const useLocationIsNeededStatus = (): boolean => {
+  const [isLocationNeeded, setIsLocationNeeded] = useState(false)
+
+  useEffect(() => {
+    const determineLocationIsNeeded = async () => {
+      const supportsLocationlessScanning = await determineSupportsLocationlessScanning()
+      setIsLocationNeeded(!supportsLocationlessScanning)
+    }
+
+    determineLocationIsNeeded()
+  }, [])
+
+  return isLocationNeeded
+}
+
+const determineSupportsLocationlessScanning = async (): Promise<boolean> => {
+  if (Platform.OS === "android") {
+    return doesDeviceSupportLocationlessScanning()
+  } else {
+    return true
```

Comment:
I'm confused as to why we can (have to?) return `boolean` here when the function says it returns `Promise<boolean>`.